src/Octoshift/AdoApi.cs

```diff
@@ -57,6 +57,31 @@ public virtual async Task<string> GetOrgOwner(string org)
             return $"{ownerName} ({ownerEmail})";
         }
 
+        public virtual async Task<DateTime> GetLastPushDate(string org, string teamProject, string repo)
+        {
+            var url = $"{_adoBaseUrl}/{org}/{teamProject}/_apis/git/repositories/{repo}/pushes?$top=1&api-version=7.1-preview.2";
+            var response = await _client.GetAsync(url);
+
+            var data = JObject.Parse(response);
+            var pushDate = data.ContainsKey("value") && data["value"].Any() ? (string)data["value"].First()["date"] : DateTime.MinValue.ToString();
```

Comment:
CodeQL's complain about this is legit (although in our case can be considered micro-optimization), it can be rewritten like the following just to make CodeQL happy 😄 :
```suggestion
            var pushDate = data.TryGetValue("value", out var values) && values.Any() ? (string)values.First()["date"] : DateTime.MinValue.ToString();
```